kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/net2/status/NetworkStatusServiceAdapter.java

```diff
@@ -72,44 +77,49 @@ public Optional<GwtNetInterfaceConfig> fillWithStatusProperties(String ifname,
-            gwtConfig.setHwState(networkInterface.getState().name());
-        }
-        if (networkInterface.getType() != null) {
-            gwtConfig.setHwType(networkInterface.getType().name());
-        }
-        gwtConfig.setHwAddress(NetUtil.hardwareAddressToString(networkInterface.getHardwareAddress()));
-        gwtConfig.setHwName(networkInterface.getName());
-        gwtConfig.setHwDriver(networkInterface.getDriver());
-        gwtConfig.setHwDriverVersion(networkInterface.getDriverVersion());
-        gwtConfig.setHwFirmware(networkInterface.getFirmwareVersion());
-        gwtConfig.setHwMTU(networkInterface.getMTU());
-        if (networkInterface.getUsbDevice() != null) {
-            gwtConfig.setHwUsbDevice(networkInterface.getUsbDevice().getUsbDevicePath());
-        }
+            NetworkInterfaceStatus networkInterfaceStatus) {
+
+        gwtConfig.setHwState(networkInterfaceStatus.getState().name());
+        gwtConfig.setHwType(networkInterfaceStatus.getType().name());
+        gwtConfig.setHwAddress(NetUtil.hardwareAddressToString(networkInterfaceStatus.getHardwareAddress()));
+        gwtConfig.setHwName(networkInterfaceStatus.getName());
+        gwtConfig.setHwDriver(networkInterfaceStatus.getDriver());
+        gwtConfig.setHwDriverVersion(networkInterfaceStatus.getDriverVersion());
+        gwtConfig.setHwFirmware(networkInterfaceStatus.getFirmwareVersion());
+        gwtConfig.setHwMTU(networkInterfaceStatus.getMtu());
+        networkInterfaceStatus.getUsbNetDevice()
+                .ifPresent(usbDevice -> gwtConfig.setHwUsbDevice(usbDevice.getUsbDevicePath()));
+
+        logger.debug("GWT common state properties for interface {}:\\n{}\\n", gwtConfig.getName(),
+                gwtConfig.getProperties());
     }
 
     private void setIpv4DhcpClientProperties(GwtNetInterfaceConfig gwtConfig,
-            NetInterface<NetInterfaceAddress> networkInterface) {
+            NetworkInterfaceStatus networkInterfaceInfo) {
 
         String ipConfigMode = gwtConfig.getConfigMode();
         if (isDhcpClient(ipConfigMode)) {
             /*
              * An interface can have multiple active addresses, we select just the first
              * one. This is a limit of the current GWT UI.
              */
-            if (!networkInterface.getNetInterfaceAddresses().isEmpty()) {
-                NetInterfaceAddress address = networkInterface.getNetInterfaceAddresses().get(0);
-                gwtConfig.setIpAddress(address.getAddress().getHostAddress());
-                gwtConfig.setGateway(address.getGateway().getHostAddress());
-                gwtConfig.setSubnetMask(address.getNetmask().getHostAddress());
-                gwtConfig.setReadOnlyDnsServers(prettyPrintDnsServers(address.getDnsServers()));
-            }
+            networkInterfaceInfo.getInterfaceIp4Addresses().ifPresent(address -> {
+                gwtConfig.setIpAddress(address.getAddresses().get(0).getAddress().getHostAddress());
```

Comment:
Are we sure that the address list is never empty? `address.getAddresses().get(0)` will never produce an index out of bound exception?